### ISSUE TITLE
Shy preloading/activation

### ISSUE THREAD

**Issue by ghost** [pisi/Reel#224]
If we have a lot of reel in the same page, it could help to limit to only the reel a user want to play.

Also, if the reel is a long sequence, we could only load the frames on demand to limit the number of frames loaded by turning preload off.

**Comment by pisi** [comment]
This kind of matches one of the [next](http://geel360.org#next) points on the site:

> Explicit activation to lower the bandwidth cost when there's no action from the user.

Although on the site it says "implicit", "explicit" is the correct word ;)

I'm thinking about adding some Boolean option, which will enable "on-demand mode", in which it will begin the preloading process only after either image was clicked or an event is triggered. Does that sound good?

**Comment by ghost** [comment]
This sounds pretty good! :+1:

**Comment by pisi** [comment]
OK, so as this was fairly straight forward, is already merged into [**development** branch](/pisi/Reel/tree/development).

I've added a **new option `shy`**, which by default is `false` and Reel initializes and preloads right away. When `true` on the other hand, it will just crack the surface with the initialization and won't trigger the setup chain of events thus effectively preventing the preload. It will wait for user to click on the image, which will trigger `"setup"` event and finish building and preloading the instance. You can of course trigger the `"setup"` event by yourself from wherever.

Have had a productive night :)

**Comment by pisi** [comment]
Ready for release

**Comment by pisi** [comment]
Released today as part of [v1.3.0](/pisi/Reel/tree/v1.3.0)
